pydantic_ai_slim/pydantic_ai/agent/__init__.py

```diff
@@ -1479,6 +1482,78 @@ def _set_sampling_model(toolset: AbstractToolset[AgentDepsT]) -> None:
 
+    def to_web(
+        self,
+        *,
+        models: Sequence[models.Model | models.KnownModelName | str]
+        | dict[str, models.Model | models.KnownModelName | str]
+        | None = None,
+        builtin_tools: list[AbstractBuiltinTool] | None = None,
+        deps: AgentDepsT = None,
+        model_settings: ModelSettings | None = None,
+        instructions: str | None = None,
+    ) -> Starlette:
+        """Create a Starlette app that serves a web chat UI for this agent.
+
+        This method returns a pre-configured Starlette application that provides a web-based
+        chat interface for interacting with the agent. The UI is downloaded and cached on
+        first use, and includes support for model selection and builtin tool configuration.
+
+        The returned Starlette application can be mounted into a FastAPI app or run directly
+        with any ASGI server (uvicorn, hypercorn, etc.).
+
+        Note that the `deps` and `model_settings` will be the same for each request.
+        To provide different `deps` for each request use the lower-level adapters directly.
+
+        Args:
+            models: Models to make available in the UI. Can be:
+                - A sequence of model names/instances (e.g., `['openai:gpt-5', 'anthropic:claude-sonnet-4-5']`)
+                - A dict mapping display labels to model names/instances
+                  (e.g., `{'GPT 5': 'openai:gpt-5', 'Claude': 'anthropic:claude-sonnet-4-5'}`)
+                If not provided, uses the agent's configured model.
+                Builtin tool support is automatically determined from each model's profile.
+            builtin_tools: Builtin tools to make available. If not provided, uses the
+                agent's configured builtin tools. Tool labels in the UI are derived
+                from the tool's `label` property.
+            deps: Optional dependencies to use for all requests.
+            model_settings: Optional settings to use for all model requests.
+            instructions: Optional extra instructions to pass to each agent run.
+
+        Returns:
+            A configured Starlette application ready to be served (e.g., with uvicorn)
+
+        Example:
+            ```python
+            from pydantic_ai import Agent
+            from pydantic_ai.builtin_tools import WebSearchTool
+
+            agent = Agent('openai:gpt-5', builtin_tools=[WebSearchTool()])
+
+            # Simple usage - uses agent's model and builtin tools
+            app = agent.to_web()
+
+            # Or provide additional models for UI selection
+            app = agent.to_web(models=['openai:gpt-5', 'anthropic:claude-sonnet-4-5'])
+
+            # Then run with: uvicorn app:app --reload
+            ```
+        """
+        from ..ui._web import ModelsParam, create_web_app
+
+        # weird ternary for typing purposes
+        resolved_models: ModelsParam = models or (self._model and [self._model])
```

Comment:
I'd prefer to only pass the additional models to `create_web_app(models=...)`, not `self._model` as well, if we could make it work like that (so that `create_web_app` checks if there are any custom models etc)

Reply:
yeah 1